sygnal.yaml.sample

```diff
@@ -123,6 +123,24 @@ http:
   #
   port: 5000
 
+## Proxying for outgoing connections #
+#
+proxy:
+  # Specify whether or not to use a proxy.
+  # Currently only HTTP proxies with CONNECT capability are supported.
+  #
+  enabled: false
+
+  # Specify an address and port to use for connecting to the HTTP proxy.
+  #
+  address: '127.0.0.1:8080'
+
+  # If your proxy uses HTTP proxy basic authentication, specify the username
+  # and password here.
+  #
+  #username:
+  #password:
+
```

Comment:
I don't strongly oppose this, but is there a reason to prefer it over the semi-standard `https_proxy` env var?

Reply:
Good point.

On one hand, you might want the Sygnal config all in one place. (I'd have to use a shell script to launch Sygnal with a proxy then...)

On the other hand, you might want the system's config in one place...

I don't know if there are security implications of putting secrets in an env var? Feels slightly dodgy but that's probably just because I wouldn't at the command-line...

I'll trust whatever you think is best but that's my take on it.

Reply:
@michaelkaye : do you have any opinions on this?

Reply:
So I've definitely seen a lot of situations where the proxy is run with the envvar in place - it's not something particularly weird to see.

The only time it might not be a good idea is if you have multiple proxy configurations for different pieces of the system, eg one for talking to google and one for talking to apple.

But practically, one http_proxy envvar is fine.

Reply:
it would be good to take the format of the http proxy envvar as well; 

http://user:pass@host:port/

Reply:
Thanks! I've put this in now